performance-tests/bench_2/node_controller/main.cpp

```diff
@@ -809,10 +809,11 @@ int run_cycle(
   }
 
   // Run Workers and Wait for Them to Finish
-  if (!worker_manager.run_workers(report_writer_impl)) {
+  if (!worker_manager->run_workers(report_writer_impl)) {
     std::cerr << "Running workers failed (likely because we received a SIGINT)" << std::endl;
     return 1;
   }
+  worker_manager.reset();
```

Comment:
Shouldn't this be `worker_manager->reset()`?

Reply:
No, it's resetting the shared pointer itself (destroying the WorkerManager in order to end the signal trapping). WorkerManager doesn't have a reset() method.

Reply:
I see.